routing/control_tower.go

```diff
@@ -107,43 +111,67 @@ func (p *controlTower) RegisterAttempt(paymentHash lntypes.Hash,
 func (p *controlTower) Success(paymentHash lntypes.Hash,
 	preimage lntypes.Preimage) error {
 
-	route, err := p.db.Success(paymentHash, preimage)
+	_, err := p.db.Success(paymentHash, preimage)
+	if err != nil {
+		return err
+	}
+
+	payment, err := p.db.FetchPayment(paymentHash)
```

Comment:
Can something happen in between these two calls?

Reply:
modified `Success` and `Fail` to return the `MPPayment` so that they are updated and read atomically